packages/common/src/GRPCClient.ts

```diff
@@ -305,6 +305,78 @@ export default class ConcordiumNodeClient {
             .response;
         return Buffer.from(response.value).toString('hex');
     }
+
+    /**
+     * Gets a stream of finalized blocks.
+     *
+     * @param abortSignal an AbortSignal to close the stream. Note that the
+     * stream does not close itself as it is infinite, so usually you'd want
+     * to provide this parameter.
+     * @returns An AsyncIterator stream of finalized blocks.
+     */
+    getFinalizedBlocks(
+        abortSignal?: AbortSignal
+    ): AsyncIterable<v1.FinalizedBlockInfo> {
+        const opts = { abort: abortSignal };
+        const blocks = this.client.getFinalizedBlocks(v2.Empty, opts).responses;
+        return mapAsyncIterable(blocks, translate.commonBlockInfo);
+    }
+
+    /**
+     * Gets a stream of blocks. To get a stream of only finalized blocks
+     * use `getFinalizedBlocks()` instead.
+     *
+     * @param abortSignal an AbortSignal to close the stream. Note that the
+     * stream does not close itself as it is infinite, so usually you'd want
+     * to provide this parameter.
+     * @returns An AsyncIterator stream of blocks.
+     */
+    getBlocks(abortSignal?: AbortSignal): AsyncIterable<v1.ArrivedBlockInfo> {
+        const opts = { abort: abortSignal };
+        const blocks = this.client.getBlocks(v2.Empty, opts).responses;
+        return mapAsyncIterable(blocks, translate.commonBlockInfo);
+    }
+
+    /**
+     * Waits until given transaction is finalized.
+     *
+     * @param transactionHash a transaction hash as a bytearray.
+     * @param timeoutTime the number of milliseconds until the function throws error.
+     * @returns A blockhash as a byte array.
+     */
+    async waitForTransactionFinalization(
+        transactionHash: HexString,
+        timeoutTime?: number
+    ): Promise<HexString> {
+        if (timeoutTime) {
+            setTimeout(() => {
+                throw Error('Function timed out.');
+            }, timeoutTime);
+        }
+
+        const abortController = new AbortController();
+        const blockStream = this.getFinalizedBlocks(abortController.signal);
+
+        const response = await this.getBlockItemStatus(transactionHash);
```

Comment:
`getBlockItemStatus` throws if the hash does not belong to known transaction? :thinking:

Reply:
It is probably not a problem in actual use cases, but it does make this test unusable:
```
test('waitForTransactionFinalization instant reject', async () => {
    const hash = '3de823b876d05cdd33a311a0f84124079f5f677afb2534c4943f830593edc650';
    await expect(clientV2.waitForTransactionFinalization(
        hash,
        1
    )).rejects.toEqual('Function timed out.')
})
```